backend/monitoring/dashboard.py

```
@@ -746,5 +1108,223 @@ async def websocket_endpoint(websocket: WebSocket):
+            social_engine_status["error"] = str(e)
+        
+        # Test SocialMediaValidator availability
+        validator_status = {"available": False, "error": None}
+        try:
+            validator = SocialMediaValidator()
+            validator_status["available"] = True
+        except Exception as e:
+            validator_status["error"] = str(e)
+        
+        # Get recent social media metrics
+        conn = await asyncpg.connect(DATABASE_URL)
```

Ruff (0.12.2)

Undefined name `asyncpg`

(F821)